test/unit/context/test-node-components.js

```diff
@@ -591,5 +597,403 @@ describes.realWin('ContextNode - components', {}, (env) => {
         expect(effectSpy).to.be.calledOnce; // no change.
       });
     });
+
+    describe('child props', () => {
+      let ComponentSelfProps;
+      let ComponentParentProps;
+      let sibling1Stub;
+      let parentStub;
+      let grandparentStub;
+
+      beforeEach(() => {
+        ComponentSelfProps = (unusedNode, input) => {
+          const setChildProp = useSetChildProp();
+          const removeChildProp = useRemoveChildProp();
+          if (input) {
+            setChildProp(Concat, input);
+          } else {
+            removeChildProp(Concat);
+          }
+        };
+        ComponentParentProps = (unusedNode, input) => {
+          const setChildProp = useSetChildProp();
+          const removeChildProp = useRemoveChildProp();
+          if (input) {
+            setChildProp(Concat, input, parent.node);
+          } else {
+            removeChildProp(Concat, parent.node);
+          }
+        };
+
+        sibling1Stub = sandbox.stub();
+        parentStub = sandbox.stub();
+        grandparentStub = sandbox.stub();
+        sibling1.values.subscribe(Concat, sibling1Stub);
+        parent.values.subscribe(Concat, parentStub);
+        grandparent.values.subscribe(Concat, grandparentStub);
+        clock.runAll();
+        [sibling1Stub, parentStub, grandparentStub].forEach((stub) => {
+          stub.resetHistory();
+        });
+      });
+
+      it('should set props in components', () => {
+        setComponent(grandparent.node, ComponentSelfProps, 'A');
+        clock.runAll();
+        expect(sibling1Stub).to.be.calledOnce.calledWith('A');
+        expect(parentStub).to.be.calledOnce.calledWith('A');
+        expect(grandparentStub).to.be.calledOnce.calledWith('A');
+
+        setComponent(grandparent.node, ComponentParentProps, 'B');
```

Comment:
IMO the `ComponentParentProps` and `ComponentSelfProps` can be somewhat confusingly named, especially since it is used with a subject called the `grandparent` as the "self", making the "parent" reference really mean the child of the `grandparent`. Is that the appropriate way to be reading this?

Reply:
Maybe it could be more clear to name them by depth instead of by role?

Reply:
Agree. Renamed.